### PR TITLE
[NEW] Support for offline messages

### LINKED ISSUE #1153: [NEW] Allow user to send offline messages and let user try again once there's connection available
## Description

- Have a failed state for messages;
- Let the user type/send messages while offline;
- Keep the message as temporary;
- Show something to let user know that message wasn't sent yet (and will be in the future);
- Add some button so user is able to "force try" sending it;

### PR DESCRIPTION
**Pull request by cardoso**
@RocketChat/ios 

Preview:

|Failing|Resend|Resent
|-|-|-|
||||

Closes #1153

### CODE REVIEW

File: Rocket.Chat/Resources/pl.lproj/Localizable.strings [hunk 1]
```diff
@@ -133,6 +133,8 @@
 "chat.message.actions.delete.confirm.title" = "Usunąć wiadomość?";
 "chat.message.actions.delete.confirm.message" = "Czy na pewno usunąć tą wiadomość? Tej czynności nie można cofnąć.";
 "chat.message.actions.delete.confirm.yes" = "Tak, usuń";
+"chat.message.actions.resend" = "Resend"; //TODO
```
Comment (artrmz) [inline Rocket.Chat/Resources/pl.lproj/Localizable.strings]: "chat.message.actions.resend" = "Wyślij ponownie";
"chat.message.actions.resend_all" = "Wyślij wszystkie ponownie";

Comment (cardoso) [reply Rocket.Chat/Resources/pl.lproj/Localizable.strings]: Thanks!! 😃

**Comment by rafaelks** [comment]
@cardoso Could you please resolve the conflicts?

**Comment by cardoso** [comment]
Fixed conflicts @rafaelks

**Comment by rafaelks** [comment]
As a clock icon, you can use this one @cardoso:

clock
clock 2x
clock 3x

**Comment by cardoso** [comment]
@rafaelks 

new changes: show clock when temporary, discard failed message

**Comment by cardoso** [comment]
@rafaelks 

new changes: reduced status icon size by 2px width 2px height

**Comment by rafaelks** [comment]
@cardoso Looking great! Another thing that I forgot to add to this feature requirement: if I close the app while there are temporary message, it would be super nice if I could resend the messages once I open the app again. Today, you can't do anything about it, because the app won't try to send them again, right? What do you think?

**Comment by cardoso** [comment]
@rafaelks did you test it? I think I covered that case here. 🤔 

https://github.com/RocketChat/Rocket.Chat.iOS/pull/1270/files#diff-914164611cb69f4158d88d4fb8927685R93

https://github.com/RocketChat/Rocket.Chat.iOS/pull/1270/files#diff-b058d63847ed636d85a604fbb258474bR192

**Comment by rafaelks** [comment]
@cardoso You're right!!! Perfect! 🎉

**Pull request merged**